src/InteractiveWindow/Editor/InteractiveWindow.cs

```diff
@@ -571,85 +569,32 @@ private void CaretPositionChanged(object sender, CaretPositionChangedEventArgs e
 
         #region Active Code and Standard Input
 
-        // TODO: What happens if multiple non-UI threads call this method? (https://github.com/dotnet/roslyn/issues/3984)
         TextReader IInteractiveWindow.ReadStandardInput()
         {
             // shouldn't be called on the UI thread because we'll hang
             RequiresNonUIThread();
+            return ReadStandardInputAsync().GetAwaiter().GetResult();
```

Comment:
I understand this will propagate the original exception, rather than an ```AggregateException```. http://stackoverflow.com/questions/17284517/is-task-result-the-same-as-getawaiter-getresult